go/vt/vttablet/tabletmanager/state_change.go

```diff
@@ -218,6 +218,18 @@ func (agent *ActionAgent) changeCallback(ctx context.Context, oldTablet, newTabl
 					disallowQueryReason = "master tablet with filtered replication on"
 				}
 			}
+			if newTablet.Type == topodatapb.TabletType_REPLICA {
+				replicationDelay, healthErr := agent.HealthReporter.Report(true, true)
```

Comment:
Generally this makes a bunch of sense. I'm mostly amused as my notes on #4426 suggests I'd tried this or similar and couldn't get it working. Though possibly I went the route of `agent.runHealthCheckLocked()` which contains a `Report` instead of calling directly. Do you know if that would make a difference here?

If not, should we call it so that we get the lag data recorded as part of the history & a report sent to observers?

Reply:
I tried this, and it had the unexpected side effect of changing a REPLICA tablet to UNKNOWN in some of the unit tests. Need to dig deeper.

Reply:
I discussed this with @sougou and he's of the opinion that at this level in the code we should be calling low-level functions and not something like `runHealthCheckLocked`.

Reply:
Oh nice. This is so close to what we tried. I remember running into the same issue you mention of `UNKNOWN`. I think I also tried `runHealthCheckLocked`